tests/subtractions/test_api.py

```diff
@@ -144,361 +108,555 @@ async def test_get_from_job(fake: DataFaker, spawn_job_client, snapshot_recent):
+    resp = await client.patch(f"/subtractions/{subtraction.id}", data)
 
     assert resp.status == 200
-    assert await resp.json() == snapshot
-    assert await mongo.subtraction.find_one() == snapshot
+    assert await resp.json() == snapshot_recent
+    assert await mongo.subtraction.find_one() == snapshot_recent
 
 
 @pytest.mark.parametrize("exists", [True, False])
-async def test_delete(
+async def test_delete_as_user(
     exists: bool,
-        fake: DataFaker,
-    resp_is,
-    mongo: Mongo,
+    fake: DataFaker,
     spawn_client: ClientSpawner,
-    tmp_path,
 ):
     client = await spawn_client(
         authenticated=True,
         permissions=[Permission.modify_subtraction],
     )
 
-    get_config_from_app(client.app).data_path = tmp_path
-
     if exists:
         user = await fake.users.create()
-        job = await fake.jobs.create(user)
+        upload = await fake.uploads.create(
+            user=user,
+            upload_type=UploadType.subtraction,
+        )
+        subtraction = await fake.subtractions.create(
+            user=user,
+            upload=upload,
+            finalized=True,
+        )
 
-        await mongo.subtraction.insert_one(
-            {
-                "_id": "foo",
-                "name": "Foo",
-                "deleted": False,
-                "ready": False,
-                "nickname": "Foo Subtraction",
-                "user": {"id": user.id},
-                "job": {"id": job.id},
-            },
+        resp = await client.delete(f"subtractions/{subtraction.id}")
+
+        assert resp.status == 204
+    else:
+        resp = await client.delete("subtractions/does_not_exist")
+        assert resp.status == 404
+
+
+class TestUploadSubtractionFileAsJob:
+    """
+    Test suite for uploading subtraction files.
+
+    Fixtures:
+        - setup: Prepares the test environment, creating necessary directories and test data.
+
+    Test cases:
+        - test_not_found: Verifies the API response when attempting to upload to a non-existent subtraction.
+        - test_create: Checks successful creation of a subtraction file.
+        - test_subtraction_file_invalid_name_error: Ensures proper handling of invalid file names.
+        - test_subtraction_file_name_conflict_error: Verifies handling of file name conflicts.
+    """
+
+    VALID_SUBTRACTION_FILE_NAME = "subtraction.1.bt2"
+    INVALID_SUBTRACTION_FILE_NAME = "reference.1.bt2"
+
+    @dataclass
+    class Context:
+        file_directory: Path
+        subtraction_directory: Path
+
+    @pytest.fixture()
+    def ctx(
+        self,
+        tmp_path: Path,
+    ):
+        test_dir: Path = tmp_path / "files"
+        test_dir.mkdir(exist_ok=True)
+
+        file_directory: Path = test_dir
+        subtraction_directory: Path = tmp_path / "subtractions"
+
+        return self.Context(
+            file_directory=file_directory,
+            subtraction_directory=subtraction_directory,
         )
```

Comment:
This is only used in `test_create` and doesn't need to be a fixture.

---
In fact, I don't think it needs to exist at all.

When you use `spawn_job_client` an application server is created with it own temporary data directory (same as the `--data-path` configuration value used when starting the server.

The `data_path` is accessible using the eponymous fixture. This means the following can be simplified:
```py
# Initial
assert os.listdir(
    ctx.subtraction_directory / subtraction.id,
) == [
    self.VALID_SUBTRACTION_FILE_NAME,
]

# With ``data_path`` fixture.
assert os.listdir(data_path / "subtractions" / subtraction.id) == [
    self.VALID_SUBTRACTION_FILE_NAME,
]
```

Let me know if I am missing some nuance in this case.

Reply:
Yes, sorry. An artifact of when I thought I needed it in all tests. removing.